src/components/databank/index.tsx

```diff
@@ -0,0 +1,451 @@
+import * as React from "react";
+import * as actions from "../../actions/";
+import BasicLayout from "../generic/BasicLayout";
+import "../../styles/common.css";
+import { KeycloakInstance } from "keycloak-js";
+import { List, Breadcrumb, Button, Icon, Modal, FormButton, Input, Select } from "semantic-ui-react"
+import TiedostoIkoni from "../../gfx/tiedostoikoni.svg";
+import KuvaTiedosto from "../../gfx/kuva.svg";
+import PdfIkoni from "../../gfx/pdfikoni.svg";
+import Api, { SharedFile, FileType } from "pakkasmarja-client";
+import { StoreState } from "src/types";
+import { Dispatch } from "redux";
+import { connect } from "react-redux";
+import strings from "src/localization/strings";
+
+/**
+ * Component props
+ */
+interface Props {
+  authenticated: boolean,
+  keycloak?: KeycloakInstance,
+}
+
+/**
+ * Component state
+ */
+interface State {
+  /**
+   * Path to the current location in databank
+   */
+  path: string;
+  /**
+   * Files and folders that will be rendered
+   */
+  sharedFiles: SharedFile[];
+  /**
+   * Whether or not add new modal is open
+   */
+  addNewodalOpen: boolean;
+  /**
+   * Contains data of new shared file
+   */
+  newSharedFile: {name?:string,type?:string,file?:File};
```

Comment:
Change to Partial class for clearer code
```suggestion
  newSharedFile: Partial<SharedFile>;
```